### ISSUE TITLE
Z-Offset only for the 1st layer

### ISSUE THREAD

**Issue by cnaslain** [fieldOfView/ZOffsetPlugin#27]
Hi,
I think the actual behavior of this plugin is to apply a Z-Offset for all the print.
Would it be possible to use/modify this plugin to apply a Z-Offset for only the 1st layer (or a defined number of layers)?
Kind regards
Chris

**Comment by fieldOfView** [comment]
That would create a gap (weakness) between the first and the second layer. Here's a visualisation in ASCII art:
```
A:     B:     C:
---    ___    ---
---    ___    ---
---    ___    ---
---    ___    ___
```
A is the normal gcode. B is what the Z-Offset plugin does. C is what you propose. Notice the distance between the bottom layer and the first layer above that.

**Comment by cnaslain** [comment]
I understand what you mean. Thanks for the ASCII art.

Let imagine the initial Z offset is -3.00 with a layer height of 0.20. To squash the 1st layer so it has a better grip with the board, I shift the Z offset using baby stepping by a value of -0.10 to -3.10, then at the end of the 1st layer I shift the z offset from +0.10 to -3.00 to reset the level to the origin.

Such manipulation is ultimately what you describe in ASCII art, isn't it?

Is this nonsense in 3D printing? Maybe a bad workaround to try and fix a stickiness problem?

I did a small print test of a small cylinder with the original plugin gcode:
```
...
;LAYER_COUNT:5
;LAYER:0
M107
G0 F5400 X111.79 Y115.005 Z0.10 ;adjusted by z offset
G92 Z0.2 ;consider this the original z before offset
;TYPE:SKIRT
...
;MESH:NONMESH
G0 F300 X148.48 Y125.711 Z0.4
...
;LAYER:1
...
```

The I did another test with adding a G92 just before the layer 1 move:
```
...
;LAYER_COUNT:5
;LAYER:0
M107
G0 F5400 X111.79 Y115.005 Z0.10 ;adjusted by z offset
G92 Z0.2 ;consider this the original z before offset
;TYPE:SKIRT
...
;MESH:NONMESH
G92 Z0.0 ;Back to original z coordinates system before offset <------------ ADDED -----------
G0 F300 X148.48 Y125.711 Z0.4
...
;LAYER:1
...
```

With the original code, at the beginning of the 2nd layer, the nozzle is too low and scratches on the 1st.
Adding the `G92 Z0.0` before the head movement of the 2nd layer, I find that the impression of the 2nd layer is better.
An explanation?
